doc/source/dev/development_environment.rst

```diff
@@ -8,7 +8,9 @@ Recommended development setup
 
 Since NumPy contains parts written in C and Cython that need to be
 compiled before use, make sure you have the necessary compilers and Python
-development headers installed - see :ref:`building-from-source`.
+development headers installed - see :ref:`building-from-source`. Building
+NumPy now requires use of the C99 standard, which for some older compilers
```

Comment:
Nitpick, but delete "now" - this should just document what is needed (we've presumably mentioned this in the change-log...)

Reply:
Maybe "NumPy >= 1.17 reguires"?

Reply:
Since this for numpy development, shouldn't it simply state what is needed? I don't see the relevance for someone starting to develop to know that it was different pre-1.17.

Reply:
@mhvk NumPy 1.16 is C99 compatible, but the Windows Python 2.7 compiler won't handle it and we still flag some warnings as errors in CI for that branch. Maybe we should mention that code written for exclusively for NumPy >= 1.17 requires C99, but that 1.16 does not fully support it on account of Python 2.7 still being supported.

Reply:
Fair enough, but this is about setting up one's development environment. Do we really want to bother people with things that are relevant  for backporting to 1.16 but not for new features?

Reply:
OK, I see that point. So, maybe just replace the "now" with an "as of version 1.17" or something like it.